package.json

```
@@ -32,7 +32,7 @@
     "body-parser": "^2.0.1",
     "content-disposition": "^1.0.0",
     "content-type": "~1.0.4",
-    "cookie": "0.6.0",
+    "cookie": "^0.7.1",
```

```suggestion
    "cookie": "0.7.1",
```
I believe that we want this value to be anchored. cc: @expressjs/express-tc

---

not in 5x if I remember correctly
CC @wesleytodd

---

anyway, within v0, `^` means patches only. So you may ask yourself: do you wanna do it again if they fix another bug?

---

And they did [release 0.7.2](https://github.com/jshttp/cookie/releases/tag/v0.7.2) just a few hours ago.

---

Would love to see it landed to 4.x branch too (and released)

---

The applications are under similar organizations. We had this problem before with mismatched dependency versions unless you do the new releases in a very specific order. I have to use npm overrides for other things until they also land.

Please keep this change as is with the caret ^. It will make my life easier in the future.
Another reason is these express dependencies are probably the best I have seen adhering to the semver principles.

Thank you.

---

Overly strict version pinning (especially in libraries) tends to cause more problems than it solves [in my experience](https://github.com/auth0/ruby-auth0/pull/279).

As long as your dependencies are properly adhering to semver ([which sometimes they don't :roll_eyes:](https://github.com/rest-client/rest-client/issues/740)), there should be no downsides to allowing backwards compatible updates with a fuzzy lock ([`^` in npm](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#dependencies)).

---

Just to unblock this PR, I will merge my suggested change and move the discussion to https://github.com/expressjs/discussions/issues/279 so we can take a team decision and after that we can do a PR updating all the dependencies with caret if needed :+1:

---

Then you should have jumped to 0.7.2 as said here https://github.com/expressjs/express/pull/6017#discussion_r1790123549

Which has this fix: https://github.com/jshttp/cookie/pull/177

I guess I will need to continue using the package.json npm override.